pkg/codegen/docs/gen_test.go

```diff
@@ -644,3 +644,227 @@ func TestDecomposeDocstring(t *testing.T) {
+			Description: "Outputs for getModuleResource.",
+			Properties:  simpleProperties,
+			Type:        "object",
+		},
+	}
+
+	testPackageSpec.Functions["prov:module/overlayFunctionConstrainedLanguages:overlayFunctionConstrainedLanguages"] = schema.FunctionSpec{
+		Description:               "A module-level function.",
+		IsOverlay:                 true,
+		OverlaySupportedLanguages: []string{"python", "go", "nodejs"},
+		Inputs: &schema.ObjectTypeSpec{
+			Description: "Inputs for getModuleResource.",
+			Type:        "object",
+			Properties: map[string]schema.PropertySpec{
+				"options": {
+					TypeSpec: schema.TypeSpec{
+						Ref: "#/types/prov:module/getModuleResource:getModuleResource",
+					},
+				},
+			},
+		},
+		Outputs: &schema.ObjectTypeSpec{
+			Description: "Outputs for getModuleResource.",
+			Properties:  simpleProperties,
+			Type:        "object",
+		},
+	}
+
+	testPackageSpec.Functions["prov:module/overlayFunctionWrongLanguage:overlayFunctionWrongLanguage"] = schema.FunctionSpec{
+		Description:               "A module-level function.",
+		IsOverlay:                 true,
+		OverlaySupportedLanguages: []string{"python", "go", "nodejs", "smalltalk"},
+		Inputs: &schema.ObjectTypeSpec{
+			Description: "Inputs for getModuleResource.",
+			Type:        "object",
+			Properties: map[string]schema.PropertySpec{
+				"options": {
+					TypeSpec: schema.TypeSpec{
+						Ref: "#/types/prov:module/getModuleResource:getModuleResource",
+					},
+				},
+			},
+		},
+		Outputs: &schema.ObjectTypeSpec{
+			Description: "Outputs for getModuleResource.",
+			Properties:  simpleProperties,
+			Type:        "object",
+		},
+	}
+
+	schemaPkg, err := schema.ImportSpec(testPackageSpec, nil)
+	assert.NoError(t, err, "importing spec")
+
+	tests := []struct {
+		FunctionName                 string
+		ExpectedLangChooserLanguages string
+	}{
+		{
+			// regular function, should support all languages
+			FunctionName:                 "getModuleResource",
+			ExpectedLangChooserLanguages: "csharp,go,typescript,python,yaml,java",
+		},
+		{
+			// Regular overlay function, should support all languages
+			FunctionName:                 "overlayFunction",
+			ExpectedLangChooserLanguages: "csharp,go,typescript,python,yaml,java",
+		},
+		{
+			// overlay function with a constrained list of supported languages
+			// should support only the languages specified in OverlaySupportedLanguages
+			FunctionName:                 "overlayFunctionConstrainedLanguages",
+			ExpectedLangChooserLanguages: "python,go,typescript",
+		},
+		{
+			// overlay function with a wrong language in OverlaySupportedLanguages
+			// should filter out the wrong language
+			FunctionName:                 "overlayFunctionWrongLanguage",
+			ExpectedLangChooserLanguages: "python,go,typescript",
+		},
+	}
+
+	modules := dctx.generateModulesFromSchemaPackage(unitTestTool, schemaPkg)
+	dctx.initialize(unitTestTool, schemaPkg)
+
+	for _, test := range tests { //nolint:paralleltest
```

Comment:
Why can't these be run in parallel?

Reply:
Did some refactoring to run them in parallel 👍

Reply:
Amazing!